src/tlp_lib/EDTLP.py

```diff
@@ -141,11 +141,8 @@ def register(self, sc: SCInterface, solution: GCTLP_Encrypted_Message, commitmen
         sc.add_solution(solution, commitment)
 
     def verify(self, sc: SCInterface, i: int) -> None:
-        solution, witness, time_solved = sc.get_solution_at(i)
+        solution, witness, _ = sc.get_solution_at(i)
         commitment = sc.get_commitment_at(i)
-        time_to_solve = time_solved - sc.initial_timestamp
-        upper_bound = sc.get_upper_bound_at(i)
-        assert time_to_solve < upper_bound
         self.gctlp.verify(solution, witness, commitment)
 
     def pay(self, sc: SCInterface, i: int) -> None:
```

Comment:
if pay is now an internal method, it won't be callable from this method, because this calls `pay` in `EthereumSC.py` which calls `_contract.functions.pay`

Reply:
This is also in the next PR. Perhaps we can just review it?